cql3/restrictions/statement_restrictions.cc

```diff
@@ -595,7 +595,9 @@ void statement_restrictions::add_single_column_parition_key_restriction(const ex
     if (has_token(_partition_key_restrictions)) {
         throw exceptions::invalid_request_exception(
                 format("Columns \"{}\" cannot be restricted by both a normal relation and a token relation",
-                        join(", ", expr::get_sorted_column_defs(_partition_key_restrictions))));
+                       fmt::join(expr::get_sorted_column_defs(_partition_key_restrictions) |
+                                 boost::adaptors::transformed([](auto* p) { return static_cast<const void*>(p); }),
```

Comment:
What does this transformation do, and why is it suddenly needed?
It seems to me that it would print the columns as pointers - but why would we want to do that? We should print the column names, not pointers. What am I missing?

Reply:
> What does this transformation do, and why is it suddenly needed?
> It seems to me that it would print the columns as pointers - but why would we want to do that? We should print the column names, not pointers. What am I missing?

very good point indeed. because of the exactly the same reason, fmtlib does not format typed pointer, but it format `void*`. and before this change, what gets printed _is_ the the pointers. i don't intend to change the behavior in this change, so added the weird looking transformation. i will add this to the commit message after fixing the FTBFS -- i am not able to reproduce it so far.

Reply:
If you believe this is a pre-existing bug (which sounds very suspicious - didn't the person who added this message see it even once?), please at least add a FIXME saying this is a bug that should be fixed. Otherwise it makes this look very deliberate.

Reply:
i believe. but i am not 100% sure it's a "bug". i'd defer this to @cvybhu .

Reply:
:+1: looks like a bug to me.
It has been this way ever since 2015: 73421cb413720c1e8526f7ac1393102e4efe29ac
`test/boost/restrictions_test/token` doesn't check column names, just the message and nobody noticed.

Reply:
@cvybhu hi Jan, thanks for ack. added FIXMEs, accordingly.

Reply:
Opened #13168

Reply:
@cvybhu @nyh hi Jan and Nadav, sorry guys. i was wrong. see https://github.com/scylladb/scylladb/commit/9c4c9eff50f0d62d63f0be41b58c0f3044ea234a . i didn't realize that we have this overload, and the hypothesis of "printing a vector of pointers would format the pointer as pointer" is just wrong.

@cvybhu hi Jan, could you please close the #13168. it was a false alarm.